File: problems/advent-of-code/2019/01/02/solution.java
public class solution {
    public static void main(String[] args) {
        int sum = 0;
        for (int x : args[0].lines().mapToInt(Integer::parseInt).toArray()) {
            while (x > 0) {
                x = Math.max(0, x / 3 - 2);
                sum += x;
            }
        }
        System.out.println(sum);
    }
}

File: problems/advent-of-code/2019/01/02/solution.rs
use std::cmp::max;

fn main() {
    let mut sum: i32 = 0;
    for line in ::std::env::args().nth(1).unwrap().lines() {
        let mut x = line.parse::<i32>().unwrap();
        while x > 0 {
            x = max(0, x / 3 - 2);
            sum += x;
        }
    }
    println!("{}", sum);
}
